compiler-rt/lib/rtsan/rtsan.cpp

```diff
@@ -49,7 +50,21 @@ static auto OnViolationAction(DiagnosticsInfo info) {
   return [info]() {
     IncrementTotalErrorCount();
 
-    PrintDiagnostics(info);
+    BufferedStackTrace stack;
+    stack.Unwind(info.pc, info.bp, nullptr,
+                 /*request_fast*/ true);
+
+    StackDepotHandle handle = StackDepotPut_WithHandle(stack);
+
+    const bool is_stack_novel = handle.use_count() == 0;
+    if (UNLIKELY(is_stack_novel)) {
```

Comment:
UNLIKELY because I think in real-time operation we will very likely get into a "steady state" where the stacks are highly duplicated.

Reply:
Please add this as a comment. In general, if it's interesting enough to mention in the review, it might be interesting for readers.

Reply:
If you really want to know, you can profile the runtime and use MisExpect to vet how accurate your guess was. 
https://clang.llvm.org/docs/MisExpect.html

Also, FWIW, we typically see `__builtin_expect_with_probability` do better on performance metrics than `__builtin_expect`.

Reply:
Cool, may be worth adding a wrapper to `__builtin_expect_with_probability` to compiler-rt at some point

Reply:
How many OnViolationAction calls you expect in normal runs? I would expect 0 or a few, so `if (is_stack_novel) {` should be totally fine here.

Reply:
In a well behaved program we expect 0.

Programs with >=1 issue with halt_on_error=true, we expect exactly one (then we die) - this is the default behavior with default flags

Programs with >=1 issue, with halt_on_error=false, when the [[nonblocking]] code is called in a loop/callback (very typical in real-time programming), we expect it to be called very often. The longer the program runs, the smaller proportion the unique calls will take up (first time through the loop will be all unique, then every loop after will be almost all duplicates when the user hits similar errors)

Happy to remove this in a follow up if you want, I defer to your judgement. I think the nature of real-time code being highly repetitive led me to UNLIKELY here